mender.go

```diff
@@ -104,20 +132,61 @@ func (m mender) GetCurrentImageID() string {
 	return imageID
 }
 
-func (m *mender) updateState() error {
+func (m *mender) changeState(state MenderState) {
+	log.Infof("Mender state: %v -> %v", m.state, state)
+	m.state = state
+}
+
+func (m *mender) hasUpgrade() (bool, error) {
```

Comment:
For consistency I think it should be `hasUpdate()`, the name of boot variable is wrong as well, but this was there before...